Cilium Operator crashes when bgp-secrets-namespace not set

https://github.com/cilium/cilium/blob/10b55acc994369fde7c95de751d8de1492472910/operator/pkg/bgpv2/peer.go#L85

When `bgp-secrets-namespace` is not set, the operator enters into a continuous crash loop. The highlighted line is the culprit.

Must `bgp-secrets-namespace` be set? The docs don't state that. And also if there's no authentication between BGP peers i don't see why it has to be set

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x2aa907d]

goroutine 356 [running]:
github.com/cilium/cilium/operator/pkg/bgpv2.(*peerConfigStatusReconciler).reconcileStatus(0xc001792d20, {0x447c978, 0xc000620ae0}, {0xc0017ffd78?, 0x4b37b3?})
	/go/src/github.com/cilium/cilium/operator/pkg/bgpv2/peer.go:85 +0x3d
github.com/cilium/hive/job.(*jobOneShot).start(0xc001792d80, {0x447c978, 0xc000620ae0}, 0x3a00327463656a?, {0x4484880, 0xc001313c20}, {{{0xc0007d0b40, 0x1, 0x1}}, 0xc00097efe0, ...})
	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/oneshot.go:138 +0x4fd
created by github.com/cilium/hive/job.(*queuedJob).Start.func1 in goroutine 273
	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/job.go:126 +0x16f
```

--- a/operator/pkg/bgpv2/peer.go
+++ b/operator/pkg/bgpv2/peer.go
@@ -75,10 +75,14 @@ func registerPeerConfigStatusReconciler(in peerConfigStatusReconcilerIn) {
 		return
 	}
 
-	in.JobGroup.Add(job.OneShot(
-		"peer-config-status-reconciler",
-		u.reconcileStatus,
-	))
+	if in.SecretResource != nil {
+		// ATM the sole purpose of this reconciliation is to populate MissingAuthSecret condition,
+		// so we don't need to run it if the SecretResource is not provided (bgp-secrets-namespace is not set).
+		in.JobGroup.Add(job.OneShot(
+			"peer-config-status-reconciler",
+			u.reconcileStatus,
+		))
+	}
 }
 
 func (u *peerConfigStatusReconciler) reconcileStatus(ctx context.Context, health cell.Health) error {

